sentry/src/test/java/io/sentry/metrics/MetricsApiTest.kt

```diff
@@ -0,0 +1,212 @@
+            any(),
+            anyOrNull()
+        )
+
+        verify(aggregator).distribution(
+            anyOrNull(),
+            anyOrNull(),
+            anyOrNull(),
+            anyOrNull(),
+            any(),
+            anyOrNull()
+        )
+    }
+
+    @Test
+    fun `timestamp is not overwritten`() {
+        val aggregator = mock<IMetricsAggregator>()
+        val api = MetricsApi(object : IMetricsInterface {
+            override fun getMetricsAggregator(): IMetricsAggregator {
+                return aggregator
+            }
+
+            override fun getDefaultTagsForMetrics(): Map<String, String> = emptyMap()
+        })
+
+        api.increment("name", 1.0, null, null, 1234)
+        api.set("name", 1, null, null, 1234)
+        api.set("name", "string", null, null, 1234)
+        api.gauge("name", 1.0, null, null, 1234)
+        api.distribution("name", 1.0, null, null, 1234)
+
+        verify(aggregator).increment(
+            anyOrNull(),
+            anyOrNull(),
+            anyOrNull(),
+            anyOrNull(),
+            eq(1234),
+            anyOrNull()
+        )
+
+        verify(aggregator).set(
+            anyOrNull(),
+            eq(1),
+            anyOrNull(),
+            anyOrNull(),
+            eq(1234),
+            anyOrNull()
+        )
+
+        verify(aggregator).set(
+            anyOrNull(),
+            eq("string"),
+            anyOrNull(),
+            anyOrNull(),
+            eq(1234),
+            anyOrNull()
+        )
+
+        verify(aggregator).gauge(
+            anyOrNull(),
+            anyOrNull(),
+            anyOrNull(),
+            anyOrNull(),
+            eq(1234),
+            anyOrNull()
+        )
+
+        verify(aggregator).distribution(
+            anyOrNull(),
+            anyOrNull(),
+            anyOrNull(),
+            anyOrNull(),
+            eq(1234),
+            anyOrNull()
+        )
+    }
+
+    @Test
+    fun `tags are enriched with default tags`() {
+        val aggregator = mock<IMetricsAggregator>()
+        val api = MetricsApi(object : IMetricsInterface {
+            override fun getMetricsAggregator(): IMetricsAggregator {
+                return aggregator
+            }
+
+            override fun getDefaultTagsForMetrics(): Map<String, String> {
+                return mapOf(
+                    "release" to "1.0",
+                    "environment" to "prod"
+                )
+            }
+        })
+        api.increment("name", 1.0, null, mapOf("a" to "b"))
+
+        verify(aggregator).increment(
+            anyOrNull(),
+            anyOrNull(),
+            anyOrNull(),
+            eq(
+                mapOf(
+                    "a" to "b",
+                    "release" to "1.0",
+                    "environment" to "prod"
+                )
+            ),
+            anyOrNull(),
+            anyOrNull()
+        )
+    }
+
+    @Test
+    fun `existing environment and release tags are not overwritten`() {
```

Comment:
Shouldn't this be `existing environment and release tags are overwritten`?

Reply:
No, it's like this: when there's a release/environment in the tags, the `MetricsApi` shouldn't overwrite those.